src/main/java/com/vmware/vipclient/i18n/messages/service/CacheService.java

```diff
@@ -101,17 +99,14 @@ public void addCacheOfStatus(Map<String, String> dataMap) {
         }
     }
 
-    public List<Locale> getCachedLocales() {
-        List<Locale> locales = new ArrayList<>();
-        Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
-        if (c == null) {
-            return locales;
-        }
-        Set<String> cacheKeys = c.keySet();
-        for (String key: cacheKeys) {
-            locales.add(getLocaleByCachedKey(key));
+    public List<Locale> getSupportedLocalesFromCache() {
```

Comment:
Don't know how the locale are from cache. Does product service get data from cache?

Reply:
If you see in ProductService, it uses the cache now.

Reply:
If it is the cases. This method should belong to product service. This is the normal way to get locales. get locale request -> service -> cache-> data source.